pkg/util/overridemanager/overridemanager.go

```diff
@@ -190,7 +190,20 @@ func (o *overrideManagerImpl) getOverridersFromOverridePolicies(policies []Gener
 			resourceMatchingPolicies = append(resourceMatchingPolicies, policy)
 		}
 	}
-
+	sort.SliceStable(resourceMatchingPolicies, func(i, j int) bool {
+		implicitPriorityI := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors) == 0 {
+			implicitPriorityI = util.PriorityMatchAll
+		}
+		implicitPriorityJ := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors) == 0 {
+			implicitPriorityJ = util.PriorityMatchAll
+		}
+		if implicitPriorityI != implicitPriorityJ {
+			return implicitPriorityI > implicitPriorityJ
```

Comment:
I did a test about this:
```go
func main() {
	a := []int{1,3,2}
	sort.SliceStable(a, func(i, j int) bool {
		return a[i] > a[j]
	})
	fmt.Println(a)
}
output:
[3 2 1]
```
So if apply these policy, low priority policy will rewrite  high low priority policy, here is a example:
```yaml
# administrator.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: OverridePolicy
metadata:
  name: sample-1
spec: 
  resourceSelectors: 
  - apiVersion: webapp.my.domain/v1 
    kind: Guestbook
  overrideRules: 
  - targetCluster: 
      clusterNames:
      - member1
    overriders:
      plaintext:
      - path: /spec/size
        operator: add
        value: 4
---
# enduser.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: OverridePolicy
metadata:
  name: sample-2
spec: 
  resourceSelectors: 
  - apiVersion: webapp.my.domain/v1 
    kind: Guestbook
    name: test
  overrideRules: 
  - targetCluster: 
      clusterNames:
      - member1
    overriders:
      plaintext:
      - path: /spec/size
        operator: repleace
        value: 4
```
So the final operation is add. Is this expected?

Reply:
Maybe we should sort the priorities in reverse order, because what is applied first will be overwritten by what is applied later.

Reply:
nice!

Reply:
I thought this might introduce a problem if a low-priority op removes a value (such as annotation key), then a high-priority op change the value will fail.

We may still need to work further on the same field changes, which can be complicated.

Reply:
> Maybe we should sort the priorities in reverse order, because what is applied first will be overwritten by what is applied later.

I still think high priority should be applied first, and then low priority should be applied

As I said https://github.com/karmada-io/karmada/pull/2609#issuecomment-1328684282